bitcoin/src/address/mod.rs

```diff
@@ -1427,4 +1487,18 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn serde_address_usage_in_struct() {
+        use serde::{Serialize, Deserialize};
+
+        #[derive(Serialize, Deserialize)]
+        struct Foo<V>
+            where V: NetworkValidation,
+        {
+            #[serde(bound(deserialize = "V: NetworkValidationUnchecked"))]
+            address: Address<V>,
+        }
```

Comment:
Might make sense to actually attempt deserialize off `Foo`, just to verify that indeed it actually works.

Reply:
Solid. The PR was buggy. I had to add derives for `Serialize, Deserialize` on `NetworkChecked` and `NetworkUnchecked` however I'm a little confused as to why this wasn't needed before.

Reply:
> confused as to why this wasn't needed before

Perhaps because there was no derivation before, only manual impls.